tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/AugmentedTaggedMetricRegistry.java

```diff
@@ -0,0 +1,202 @@
+                            SafeArg.of("tagName", tagName),
+                            SafeArg.of("existing", other.tagValue),
+                            SafeArg.of("new", tagValue));
+                }
+            }
+        }
+
+        return new AugmentedTaggedMetricRegistry(delegate, tagName, tagValue);
+    }
+
+    private MetricName augment(MetricName existing) {
+        return RealMetricName.create(existing, tagName, tagValue);
+    }
+
+    @Override
+    public Map<MetricName, Metric> getMetrics() {
+        return delegate.getMetrics();
+    }
+
+    @Override
+    public void forEachMetric(BiConsumer<MetricName, Metric> consumer) {
+        delegate.forEachMetric(consumer);
+    }
+
+    @Override
+    public <T> Optional<Gauge<T>> gauge(MetricName metricName) {
+        return delegate.gauge(augment(metricName));
+    }
+
+    @Override
+    public <T> Gauge<T> gauge(MetricName metricName, Gauge<T> gauge) {
+        return delegate.gauge(augment(metricName), gauge);
+    }
+
+    @Override
+    public void registerWithReplacement(MetricName metricName, Gauge<?> gauge) {
+        delegate.registerWithReplacement(augment(metricName), gauge);
+    }
+
+    @Override
+    public Timer timer(MetricName metricName) {
+        return delegate.timer(augment(metricName));
+    }
+
+    @Override
+    public Timer timer(MetricName metricName, Supplier<Timer> timerSupplier) {
+        return delegate.timer(augment(metricName), timerSupplier);
+    }
+
+    @Override
+    public Meter meter(MetricName metricName) {
+        return delegate.meter(augment(metricName));
+    }
+
+    @Override
+    public Meter meter(MetricName metricName, Supplier<Meter> meterSupplier) {
+        return delegate.meter(augment(metricName), meterSupplier);
+    }
+
+    @Override
+    public Histogram histogram(MetricName metricName) {
+        return delegate.histogram(augment(metricName));
+    }
+
+    @Override
+    public Histogram histogram(MetricName metricName, Supplier<Histogram> histogramSupplier) {
+        return delegate.histogram(augment(metricName), histogramSupplier);
+    }
+
+    @Override
+    public Counter counter(MetricName metricName) {
+        return delegate.counter(augment(metricName));
+    }
+
+    @Override
+    public Counter counter(MetricName metricName, Supplier<Counter> counterSupplier) {
+        return delegate.counter(augment(metricName), counterSupplier);
+    }
+
+    @Override
+    public Optional<Metric> remove(MetricName metricName) {
+        return delegate.remove(augment(metricName));
+    }
+
+    /**
+     * .
+     * @deprecated not implemented
+     */
+    @Deprecated
+    @Override
+    public void addMetrics(String _safeTagName, String _safeTagValue, TaggedMetricSet _metrics) {
+        throw new UnsupportedOperationException(
+                "Operations involving transforming TaggedMetricSet are not supported, please interact with the "
+                        + "delegate directly");
+    }
```

Comment:
i'm not 100% sure what the desired behaviour is in this case- perhaps we need an AugmentedTaggedMetricSet too? Passing it straight down unmodified feels like it might surprise users.